build.gradle

```diff
@@ -97,6 +74,11 @@ pmd {
     ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
     ruleSets = []
     ignoreFailures = false
+    pmdMain {
+        excludes = [
+                '**/BuildConfig.*'
```

Comment:
could not find a more generic way to exclude the buildconfig file. I would have liked to exclude all generated classes but could not find a way to do it. Patterns like `**/generated/**/* ` did not do the trick